spec/compiler/codegen/automatic_cast.cr

```diff
@@ -260,4 +260,14 @@ describe "Code gen: automatic cast" do
       Bar.new.as(Foo).foo(1)
       )).to_i.should eq(2)
   end
+
+  it "doesn't autocast number on union (#8655)" do
+    run(%(
+      def foo(x : UInt8 | Int32, y : Float64)
```

Comment:
If the union would have been `UInt64 | Int64` and the call with `Int32`, what is the output?

ambiguous call? or any of those types are used?

Reply:
It says ambiguous call.

Reply:
ok, if there is no spec for that it should be added. Other than tat LGTM